### PR TITLE
Change default s3_throw_on_zero_files_match to true, attempt 2

### PR DESCRIPTION
**Pull request by al13n321**
### Changelog category (leave one):
- Backward Incompatible Change

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Default value of setting s3_throw_on_zero_files_match changed from false to true to avoid confusion when inadvertently using URL containing '?' characters.

Un-reverting https://github.com/ClickHouse/ClickHouse/pull/51008 (IIUC, something went wrong with the sync PR there, but I'm not sure)

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 58f1d44298faf904066d3a627ae5555c814420f7 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/65506/58f1d44298faf904066d3a627ae5555c814420f7/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Cloud fork sync (only for ClickHouse Inc. employees)</td><td>If it fails, ask a maintainer for help</td><td>⏳ pending</td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>Checks correctness of the PR's body</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Core/SettingsChangesHistory.h [hunk 1]
```diff
@@ -88,8 +88,9 @@ static const std::map<ClickHouseVersion, SettingsChangesHistory::SettingsChanges
...
               {"input_format_parquet_use_native_reader", false, false, "When reading Parquet files, to use native reader instead of arrow reader."},
-              {"hdfs_throw_on_zero_files_match", false, false, "Allow to throw an error when ListObjects request cannot match any files in HDFS engine instead of empty query result"},
-              {"azure_throw_on_zero_files_match", false, false, "Allow to throw an error when ListObjects request cannot match any files in AzureBlobStorage engine instead of empty query result"},
+              {"s3_throw_on_zero_files_match", false, true, "Report error if S3 URL with wildcards didn't match any files. Useful in particular when trying to use pre-signed URL (which is not supported) with '?' character that is inadvertently interpreted as a wildcard."},
+              {"hdfs_throw_on_zero_files_match", false, true, "Allow to throw an error when ListObjects request cannot match any files in HDFS engine instead of empty query result"},
+              {"azure_throw_on_zero_files_match", false, true, "Allow to throw an error when ListObjects request cannot match any files in AzureBlobStorage engine instead of empty query result"},
```
Comment (Avogar) [inline src/Core/SettingsChangesHistory.h:93]: Now we should move it to `24.7` as 24.6 branch was already created

**Review by Avogar — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
@al13n321, please merge with the master branch.

**Comment by CheSema** [comment]
Is it still actual?

**Comment by alexey-milovidov** [comment]
No, and it can be superseded by https://github.com/ClickHouse/ClickHouse/issues/65032

**Pull request closed without merging**